public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -262,7 +262,40 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
                 conceptService.searchConcepts(searchConceptsRequest.getQuery(), convertedConceptFilter,
                         searchConceptsRequest.getVocabularyIds(), domainIds, maxResults);
         ConceptListResponse response = new ConceptListResponse();
-        response.setItems(concepts.getContent().stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+        List<Concept> matchedConcepts = concepts.getContent();
+        List<Concept> conceptCodeMatches = new ArrayList<>();
+        for(Concept con:matchedConcepts){
+            String conceptCode = con.getConceptCode();
+            Long conceptId = con.getConceptId();
+            if(conceptCode.equals(searchConceptsRequest.getQuery())) {
+
+                conceptCodeMatches.add(con);
+
+                response.setMatchType("ConceptCode");
+
+                Concept std_concept = conceptDao.findStandardConcept(con.getConceptId());
+                response.setStandardConcept(TO_CLIENT_CONCEPT.apply(std_concept));
+
+                response.setItems(conceptCodeMatches.stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
```

Comment:
So it's possible for us to have a query string that both matches a concept code or ID and *also* appears as a keyword in standard concepts. If that happens, what do we want the UX to be?

If we want to return the other standard concepts in these cases (which seems right to me), we shouldn't return a response here... we should just tack on a non-standard concept with the standard concept info filled out and continue. You could even do this logic as a part of TO_CLIENT_CONCEPT and leave the logic in here the way it was before.

Reply:
I thought if there is a match on concept code/id we are going to return the concept matched by code/id and throw away the rest of concept_name matches

Reply:
Yeah, I'm not sure whether this is the right behavior. It may be that some codes / concepts are short and show up in other things that researchers are looking for... might be better to always do the full search, even when there's a code / ID match. @biopete WDYT?